src/web/components/ArticleHeadline.tsx

```diff
@@ -136,42 +131,74 @@ const ageWarningMargins = css`
 `;
 
 const renderHeadline = (
-    type: HeadlineType,
+    designType: DesignType,
+    isShowcase: boolean,
     headlineString: string,
     options?: {
         colour?: string;
     },
 ) => {
-    switch (type) {
-        case 'basic':
-        case 'bold':
-        case 'light':
-        case 'underlined': {
+    switch (designType) {
```

Comment:
I'm in two minds, but there is the [designTypeDefault helper function](https://github.com/guardian/dotcom-rendering/blob/0832b716b8ca59cfaa982224a04e6fd0bbb5bf29/src/lib/designTypes.ts#L22) for this case:

```js
const default: DesignTypesObj = designTypeDefault(
         <h1 className={cx(standardFont, standardPadding)}>
                    {curly(headlineString)}
                </h1>
    );

// Extend defaults with custom  for some designTypes
    const headlineByDesignType: DesignTypesObj = {
        ...default,
        AdvertisementFeature: <h1 className={cx(standardFont, standardPadding)}>
                    {curly(headlineString)}
                </h1>,
		Feature: <h1
                    className={cx(
                        boldFont,
                        standardPadding,
                        colourStyles(options && options.colour),
                    )}
                >
                    {curly(headlineString)}
                </h1>
    }
	...etc
}
    return headlineByDesignType[designType];
```

Reply:
The advantage to using `switch` is there's a lot of grouping, which is visually very clear with switch but wouldn't be possible with `designTypeDefault`. For example, here's how `Standfirst` will look:

```
    switch (designType) {
        case 'Comment':
        case 'GuardianView':
        case 'Feature':
        case 'Recipe':
        case 'Review':
            return css`
                ${headline.xxsmall({
                    fontWeight: 'light',
                })};
            `;
        case 'Immersive':
            return css`
                ${headline.xsmall({
                    fontWeight: 'light',
                })};
            `;
        case 'Media':
        case 'SpecialReport':
        case 'MatchReport':
        case 'AdvertisementFeature':
        case 'GuardianLabs':
        case 'Quiz':
        case 'Article':
        case 'Live':
        case 'Analysis':
        case 'Interview':
        default:
            return css`
                ${headline.tiny({
                    fontWeight: 'bold',
                })};
                line-height: 20px;
            `;
    }
```

There's not really a default, instead you have 2 possible style groups and an exception.